js/views/modals/Settings/Advanced.js

```diff
@@ -0,0 +1,103 @@
+import app from '../../../app';
+import loadTemplate from '../../../utils/loadTemplate';
+import baseVw from '../../baseVw';
+import $ from 'jquery';
+
+const siblingsToggle = '[data-next-siblings-toggle]';
+const siblingsToggleInputs = `${siblingsToggle} input[type="radio"]`;
+
+export default class extends baseVw {
+  constructor(options = {}) {
+    super({
+      className: 'settingsAdvanced',
+      ...options,
+    });
+
+    this.settings = app.settings.clone();
+
+    this.listenTo(this.settings, 'sync', () => app.settings.set(this.settings.toJSON()));
+  }
+
+  get events() {
+    return {
+      [`change ${siblingsToggleInputs}`]: 'handleNextSiblingsToggle',
+    };
+  }
+
+  handleNextSiblingsToggle(event) {
+    const t = event.target;
+    this.applyNextSiblingsToggle(t, t.value);
+  }
+
+  applyNextSiblingsToggle(t, val) {
+    const p = $(t).parents(siblingsToggle);
+    if (val === 'true') {
+      p.addClass('showNextSiblings');
+    } else {
+      p.removeClass('showNextSiblings');
+    }
+  }
+
+  getFormData(subset = this.$formFields) {
+    return super.getFormData(subset);
+  }
+
+  saveServer() {
+    const formData = this.getFormData();
+    this.settings.set(formData);
+    return this.settings.save();
+  }
+
+  saveLocal() {
+    const localData = this.getFormData(this.$localFields);
+    app.localSettings.set(localData);
+    return app.localSettings.save();
+  }
+
+  reportSave(save) {
+    this.trigger('saving');
+
+    if (!save) {
+      // client side validation failed
+      this.trigger('saveComplete', true);
+    } else {
+      this.trigger('savingToServer');
+
+      save
+        .done(() => {
+          this.trigger('saveComplete');
+        })
+        .fail((...args) => {
+          this.trigger('saveComplete', false, true,
+            args[0] && args[0].responseJSON && args[0].responseJSON.reason || '');
+        });
+    }
+
+    this.render();
+    const $firstErr = this.$('.errorList:first');
+    if ($firstErr.length) $firstErr[0].scrollIntoViewIfNeeded();
+  }
+
+  save() {
+    this.reportSave(this.saveLocal() && this.saveServer());
```

Comment:
(Github is driving me bonkers. It's dropping some comments and this was a big one. Anyways, I'll try again)

How you're handling the save of these two models is problematic for a couple of reasons:

1. If one fails client side validation, but the other does not, it can result in a weird state where the UI is displaying a red error message for one model, yet the other one is being persisted. To a degree, this is masked by the fact that the nature of the radio buttons for the localSettings field won't let the user fail client validation, but for future proofing reasons, I think we should only save() both models if they both pass client validation.

2. The way you're passing the save's into reportSave means only one save will be "reported" on whereas they both should be included in the "reporting" process since they will fail / succeed independently of each other.

I would suggest doing the save as follows:

```
    this.localSettings.set(this.getFormData(this.$localFields), { validate: true });
    this.settings.set(this.getFormData(), { validate: true });

    if (this.localSettings.validationError || this.settings.validationError) {
      // client side validation failed on one or both models
      this.trigger('saveComplete', true);
    } else {
      this.trigger('savingToServer');

      // let's save and monitor both save processes
      const localSave = this.localSettings.save();
      const serverSave = this.settings.save();

      $.when(localSave, serverSave)
        .done(() => {
          // both succeeded!
          this.trigger('saveComplete');
        })
        .fail((...args) => {
          // One has failed, the other may have also failed or may
          // fail or may succeed. It doesn't matter, for our purposed one
          // failure is enough for us to consider the "save" to have failed
          this.trigger('saveComplete', false, true,
            args[0] && args[0].responseJSON && args[0].responseJSON.reason || '');
        });
    }

    this.render();
    const $firstErr = this.$('.errorList:first');
    if ($firstErr.length) $firstErr[0].scrollIntoViewIfNeeded();

```